cli/azd/cmd/show.go

```diff
@@ -0,0 +1,188 @@
+			Services: make(map[string]showService, len(prj.Services)),
+		}
+
+		for name, svc := range prj.Services {
+			path, err := getFullPathToProjectForService(svc)
+			if err != nil {
+				return err
+			}
+
+			showSvc := showService{
+				Project: showServiceProject{
+					Path: path,
+					Type: showTypeFromLanguage(svc.Language),
+				},
+			}
+
+			res.Services[name] = showSvc
+		}
+
+		// Add information about the target of each service, if we can determine it (if the infrastructure has
+		// not been deployed, for example, we'll just not include target information)
+		resourceManager := infra.NewAzureResourceManager(ctx)
+		if resourceGroupName, err := resourceManager.FindResourceGroupForEnvironment(ctx, &env); err == nil {
+			for name := range prj.Services {
+				if resources, err := project.GetServiceResources(ctx, resourceGroupName, name, &env); err == nil {
+					if len(resources) == 1 {
+						resSvc := res.Services[name]
+						resSvc.Target = &showTargetArm{
+							ResourceId: resources[0].Id,
+						}
+						res.Services[name] = resSvc
+					}
+				} else {
+					log.Printf("ignoring error determining resource id for service %s: %v", name, err)
+				}
+			}
+		} else {
+			log.Printf("ignoring error determining resource group for environment %s, resource ids will not be available: %v", env.GetEnvName(), err)
+		}
+
+		return formatter.Format(res, writer, nil)
+	}
+
+	cmd := commands.Build(
+		commands.ActionFunc(action),
+		rootOptions,
+		"show",
+		"Display information about your application and its resources.",
+		nil,
+	)
+
+	output.AddOutputParam(cmd,
+		[]output.Format{output.JsonFormat, output.NoneFormat},
+		output.NoneFormat,
+	)
+
+	cmd.Hidden = true
+
+	return cmd
+}
+
+type showResult struct {
+	Services map[string]showService `json:"services"`
+}
+
+type showService struct {
+	// Project contains information about the project that backs this service.
+	Project showServiceProject `json:"project"`
+	// Target contains information about the resource that the service is deployed
+	// to.
+	Target *showTargetArm `json:"target,omitempty"`
+}
+
+type showServiceProject struct {
+	// Path contains the path to the project for this service.
+	// When 'type' is 'dotnet', this includes the project file (i.e. Todo.Api.csproj).
+	Path string `json:"path"`
+	// The type of this project. One of "dotnet", "python", or "node"
+	Type string `json:"language"`
+}
+
+type showTargetArm struct {
+	ResourceId string `json:"resourceId"`
+}
+
+func showTypeFromLanguage(language string) string {
+	switch language {
+	case "dotnet":
+		return "dotnet"
+	case "py", "python":
+		return "python"
+	case "ts", "js":
+		return "node"
+	default:
+		panic(fmt.Sprintf("unknown language %s", language))
+	}
+}
+
+// getFullPathToProjectForService returns the full path to the source project for a given service. For dotnet services,
+// this includes the project file (e.g Todo.Api.csproj). For dotnet services, if the `path` component of the configuration
```

Comment:
I think when this came up last time in a different issue, we decided that we should stick to having a single project file in a given directory, and that's how the `azure.yaml` schema is designed to work. Given that constraint, I think we should be applying this validation in our "loading project validation".

I know that you had issues with reusing this logic if we put it in the `GetProject` construction, so what if we made this part of the `DotNetFrameworkService.Initialize` logic validation? And we expose a property/method for `GetFullProjectFilePath` to be reused here. It's still correct that `dotnet` commands will not work if there are multiple project files in a given directory, it's a prereq condition that we're not currently accounting for.

Reply:
We can special-case it for `dotnet_framework_service.go` for now, but if we see the need to expose it for all languages, it might be worth adding it as part of the interface as well -- I'm not sure what makes sense for this command's output or for VS support.

Reply:
Ah. I think I understand the change better now. We're trying to handle if multiple language projects were included. I think it makes sense to add this validation somewhere in the `dotnet` loading logic, and we should refactor it so that it be reshared here for use (since we currently have tight coupling to Azure resolution during loading -- another battle for another day).